Remotion/Web/Core/UI/Controls/WebButton.cs

```diff
@@ -282,20 +283,20 @@ protected override void RenderContents (HtmlTextWriter writer)
 
       ScriptUtility.Instance.RegisterElementForBorderSpans (this, "#" + ClientID + " > *:first-child");
 
-      writer.AddAttribute (HtmlTextWriterAttribute.Class, CssClassButtonBody);
+      writer!.AddAttribute (HtmlTextWriterAttribute.Class, CssClassButtonBody);
       writer.RenderBeginTag (HtmlTextWriterTag.Span);
 
-      var text = HotkeyFormatter.FormatText (_textWithHotkey, false);
+      var text = HotkeyFormatter.FormatText (_textWithHotkey!, false); // TODO: not null assertion
 
       if (HasControls())
         base.RenderContents (writer);
       else
       {
-        bool hasIcon = _icon != null && _icon.HasRenderingInformation;
+        bool hasIcon = _icon != null && _icon.HasRenderingInformation; // TODO: inline
```

Comment:
Is there an open ticket in Roslyn for this kind of guard? Inlining the more complicated guard conditions removes readability from the code.

Reply:
am confused? can be removed without inlining the boolean?